eZ/Bundle/EzPublishCoreBundle/Command/ReindexCommand.php

```diff
@@ -61,13 +84,230 @@ protected function configure()
+                $processes[$key] = $this->getPhpProcess($generator->current(), $commit);
+                $processes[$key]->start();
+                $generator->next();
+            }
+
+            if (!empty($processes)) {
+                sleep(1);
+            }
+        } while (!empty($processes));
+
+        $progress->finish();
+    }
+
+    /**
+     * @param DateTime $since
+     * @param string $select Field to select, one of  'c.id' and 'count(c.id)'
+     *
+     * @return \Doctrine\DBAL\Driver\Statement
+     */
+    private function getStatementContentSince(DateTime $since, $select = 'c.id')
+    {
+        $q = $this->connection->createQueryBuilder()
+            ->select($select)
+            ->from('ezcontentobject', 'c')
+            ->where('c.status = :status')->andWhere('c.modified >= :since')
+            ->orderBy('c.modified')
+            ->setParameter('status', ContentInfo::STATUS_PUBLISHED, PDO::PARAM_INT)
+            ->setParameter('since', $since->getTimestamp(), PDO::PARAM_INT);
+
+        return $q->execute();
+    }
+
+    /**
+     * @param string $select Field to select, one of  'c.id' and 'count(c.id)'
+     *
+     * @return \Doctrine\DBAL\Driver\Statement
+     */
+    private function getStatementContentAll($select = 'c.id')
+    {
+        $q = $this->connection->createQueryBuilder()
+            ->select($select)
+            ->from('ezcontentobject', 'c')
+            ->where('c.status = :status')
+            ->setParameter('status', ContentInfo::STATUS_PUBLISHED, PDO::PARAM_INT);
+
+        return $q->execute();
+    }
+
+    /**
+     * @param \Doctrine\DBAL\Driver\Statement $stmt
+     * @param int $iterationCount
+     *
+     * @return \Generator Return an array of arrays, each array contains content id's of $iterationCount.
+     */
+    private function fetchIteration(Statement $stmt, $iterationCount)
+    {
+        do {
+            $contentIds = [];
+            for ($i = 0; $i < $iterationCount; ++$i) {
+                if ($contentId = $stmt->fetch(PDO::FETCH_COLUMN)) {
+                    $contentIds[] = $contentId;
+                } else {
+                    break;
+                }
+            }
+
+            yield $contentIds;
+        } while (!empty($contentId));
+    }
+
+    /**
+     * @param array $contentIds
+     * @param bool $commit
+     *
+     * @return \Symfony\Component\Process\Process
+     */
+    private static function getPhpProcess(array $contentIds, $commit)
+    {
+        $phpFinder = new PhpExecutableFinder();
+        if (!$phpPath = $phpFinder->find()) {
+            throw new \RuntimeException('The php executable could not be found, add it to your PATH environment variable and try again');
+        }
+
+        $php = escapeshellarg($phpPath);
+        $console = escapeshellarg(file_exists('bin/console') ? 'bin/console' : 'app/console');
+        $cmd = 'ezplatform:reindex --content-ids=' . implode(',', $contentIds) . ($commit ? '' : ' --no-commit');
```

Comment:
Maybe we can use`\Symfony\Component\Process\ProcessBuilder` here?  

```php
$pb = new ProcessBuilder();
$pb->setTimeout(null);
$pb->setPrefix($phpPath);
$pb->add(file_exists('bin/console') ? 'bin/console' : 'app/console');
$pb->add('ezplatform:reindex');
$pb->add('--content-ids=' . implode(',', $contentIds));
if (!$commit) {
    $pb->add('--no-commit');
}

$process = $pb->getProcess();
```
I'm not experienced with this component, so this is only suggestion 😉

Reply:
wdyt @alongosz ?

Reply:
Didn't know this one, looks more readable
+1

Reply:
Retrospective note: ProcessBuilder got deprecated in 3.4, so we should use Process instead.
https://github.com/symfony/process/blob/3.4/ProcessBuilder.php